service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -554,14 +554,27 @@ private String resolveNamespaceLocation(Namespace namespace, Map<String, String>
         .map(
             entity -> {
               if (entity.getType().equals(PolarisEntityType.CATALOG)) {
```

Comment:
Refactor suggestion: would the code be cleaner if we put these into a new method like
```
  private static String baseLocation(PolarisEntity entity) {
    if (entity.getType().equals(PolarisEntityType.CATALOG)) {
      return CatalogEntity.of(entity).getDefaultBaseLocation();
    } else {
      String baseLocation =
          entity.getPropertiesAsMap().get(PolarisEntityConstants.ENTITY_BASE_LOCATION);
      if (baseLocation != null) {
        return baseLocation;
      } else {
        return entity.getName();
      }
    }
  }
```

Reply:
sounds good. Moved this logic to `baseLocation` helper!